spec/spec_helper.rb

```diff
@@ -28,3 +28,13 @@ def reset_i18n
     Money.infinite_precision = false
   end
 end
+
+if RUBY_VERSION >= '2.4.0'
+  module Warning
+    def self.warn(message)
```

Comment:
I think there's a cleaner solution — just implement `def warn` on `Money` class within the tests. This way instead of calling `Warning.warn` a local method will be called and you don't need to rename the warning.